tools/taskcluster/tasks.py

```diff
@@ -0,0 +1,143 @@
+            }
+        )
+
+    def craft_master_task(self):
+        script = f'''
+        git fetch {self.repo_url}
+        git config advice.detachedHead false
+        git checkout {self.commit}
+        yes | sdkmanager --licenses
+        ./gradlew -PisPullRequest clean assembleSystem assembleAndroidTest lint checkstyle ktlint pmd detekt test
+        python ./tools/taskcluster/get-bitbar-token.py
+        python ./tools/taskcluster/execute-bitbar-test.py system/debug app-system-debug
+        '''
+
+        return self._craft_shell_task(
+            "Firefox for Amazon's Fire TV - Build - Master",
+            script,
+            ['secrets:get:project/mobile/firefox-tv/tokens'],
+            {
+                'public': artifact('directory', '/opt/firefox-tv/app/builds/reports')
+            }
+        )
+
+    def craft_release_task(self, tag):
+        script = f'''
+        git fetch origin --tags
+        git config advice.detachedHead false
+        git checkout {tag}
+        yes | sdkmanager --licenses
+        python tools/taskcluster/get-sentry-token.py
+        python tools/taskcluster/get-pocket-token.py
+        ./gradlew --no-daemon clean test assembleSystemRelease
+        '''
+
+        return self._craft_shell_task(
+            f'Firefox for Fire TV - Release build {tag}',
+            script,
+            ['secrets:get:project/mobile/firefox-tv/tokens'],
+            {
+                'public': artifact('directory', '/opt/firefox-tv/app/build/outputs/apk')
+            }
+        )
+
+    def _craft_shell_task(self, name, script, scopes, artifacts):
+        single_command = ' && '.join([line.strip() for line in script.split('\n') if line.strip()])
+        bash_command = [
+            '/bin/bash',
+            '--login',
+            '-cx',
+            f'export TERM=dumb && {single_command}'
+        ]
+
+        return {
+            'taskGroupId': self.task_group_id,
+            'provisionerId': 'aws-provisioner-v1',
+            'schedulerId': 'taskcluster-github',
+            'workerType': 'github-worker',
+            'created': taskcluster.stringDate(datetime.datetime.now()),
+            'deadline': taskcluster.stringDate(taskcluster.fromNow('1 day')),
+            'scopes': scopes,
+            'payload': {
+                'maxRunTime': 3600,
+                'image': 'mozillamobile/firefox-tv:2.3',
+                'command': bash_command,
+                'artifacts': artifacts,
+                'features': {
+                    'taskclusterProxy': True,
+                }
+            },
+            'metadata': {
+                'name': name,
+                'description': '',
+                'owner': self.owner,
+                'source': f'{self.repo_url}/raw/{self.commit}/.taskcluster.yml',
+            }
+        }
+
+
+def schedule_task_graph(ordered_tasks):
+    queue = taskcluster.Queue({'rootUrl': os.environ['TASKCLUSTER_PROXY_URL']})
```

Comment:
Nit: default to `https://taskcluster.net` for smoother local run.

Reply:
:open_mouth: then it'll ... work locally? Does it just use the authentication environment variables, then it's happy? Nice!

Reply:
Hmm, it's still not happy with credentials, but now it's more obvious why it's failing. :+1: to this default